core/optaplanner-constraint-drl/src/main/java/org/optaplanner/constraint/drl/DrlScoreDirectorFactoryService.java

```diff
@@ -75,8 +82,55 @@ public Supplier<AbstractScoreDirectorFactory<Solution_, Score_>> buildScoreDirec
     }
 
     @Override
-    protected DrlScoreDirectorFactory<Solution_, Score_> createScoreDirectorFactory(ScoreDirectorFactoryConfig config,
-            SolutionDescriptor<Solution_> solutionDescriptor, KieBase kieBase) {
+    public ScoreDirectorType getSupportedScoreDirectorType() {
+        return ScoreDirectorType.DRL;
+    }
+
+    private DrlScoreDirectorFactory<Solution_, Score_> buildScoreDirectorFactory(ClassLoader classLoader,
+            SolutionDescriptor<Solution_> solutionDescriptor, ScoreDirectorFactoryConfig config,
+            List<String> scoreDrlList) {
+        KieBase kieBase;
+        if (config.getGizmoKieBaseSupplier() != null) {
+            kieBase = config.getGizmoKieBaseSupplier().get();
+        } else {
+            KieHelper kieHelper = new KieHelper(PropertySpecificOption.ALLOWED)
+                    .setClassLoader(classLoader);
+            scoreDrlList.forEach(scoreDrl -> kieHelper
+                    .addResource(KieServices.get().getResources().newClassPathResource(scoreDrl, classLoader)));
+            if (!ConfigUtils.isEmptyCollection(config.getScoreDrlFileList())) {
+                for (File scoreDrlFile : config.getScoreDrlFileList()) {
+                    kieHelper.addResource(KieServices.get().getResources().newFileSystemResource(scoreDrlFile));
+                }
+            }
+            KieBaseConfiguration kieBaseConfiguration = buildKieBaseConfiguration(config, KieServices.get());
+            kieBaseConfiguration.setOption(KieBaseMutabilityOption.DISABLED); // Performance improvement.
+            try {
+                kieBase = kieHelper.build(ExecutableModelProject.class, kieBaseConfiguration);
+            } catch (Exception ex) {
+                throw new IllegalStateException("There is an error in a scoreDrl or scoreDrlFile.", ex);
+            }
+        }
+
+        if (config.isDroolsAlphaNetworkCompilationEnabled()) {
+            KieBaseUpdaterANC.generateAndSetInMemoryANC(kieBase); // Enable Alpha Network Compiler for performance.
+        }
+        return createScoreDirectorFactory(solutionDescriptor, kieBase);
+    }
+
+    private static KieBaseConfiguration buildKieBaseConfiguration(ScoreDirectorFactoryConfig config,
+            KieServices kieServices) {
+        KieBaseConfiguration kieBaseConfiguration = kieServices.newKieBaseConfiguration();
+        if (config.getKieBaseConfigurationProperties() != null) {
+            for (Map.Entry<String, String> entry : config.getKieBaseConfigurationProperties().entrySet()) {
+                kieBaseConfiguration.setProperty(entry.getKey(), entry.getValue());
+            }
+        }
+        return kieBaseConfiguration;
+    }
+
+    private DrlScoreDirectorFactory<Solution_, Score_>
+            createScoreDirectorFactory(SolutionDescriptor<Solution_> solutionDescriptor, KieBase kieBase) {
```

Comment:
This method seems like a by-product of the abstract service class refactoring. It doesn't improve readability (compare methods signature with the body). Please inline it if you agree.